src/ai/backend/manager/registry.py

```diff
@@ -1206,26 +1206,8 @@ async def enqueue_session(
 
             async def _enqueue() -> None:
```

Comment:
```suggestion
        try:
            async def _enqueue() -> None:
                async with self.db.begin_session() as db_sess:
                    if sudo_session_enabled:
                        environ["SUDO_SESSION_ENABLED"] = "1"

                    session_data["environ"] = environ
                    session_data["requested_slots"] = session_requested_slots
                    session = SessionRow(**session_data)
                    kernels = [KernelRow(**kernel) for kernel in kernel_data]
                    db_sess.add(session)
                    db_sess.add_all(kernels)
                    await db_sess.flush()

                    if dependency_sessions:
                        matched_dependency_session_ids = []
                        for dependency_id in dependency_sessions:
                            try:
                                match_info = await SessionRow.get_session(
                                    db_sess,
                                    dependency_id,
                                    access_key,
                                    allow_stale=True,
                                )
                            except SessionNotFound:
                                raise InvalidAPIParameters(
                                    "Unknown session ID or name in the dependency list",
                                    extra_data={"session_ref": dependency_id},
                                )
                            else:
                                matched_dependency_session_ids.append(match_info.id)

                            db_sess.add(
                                SessionDependencyRow(session_id=session_id, depends_on=match_info.id)
                            )

            await execute_with_retry(_enqueue)
```

Reply:
Suggestion.

Reply:
I am sorry but how is it different?? I couldn't find the point

Reply:
@fregataa Sorry, I was trying to merge two `if-statement` blocks into a single block and remove `matched_dependency_session_ids`, but please forget this 🥲